### PR TITLE
refactor(gitlab): add conn id to project

### PR DESCRIPTION
**Pull request by warren830**
# Summary

Refactor gitlab project tasks with connection_id
Add e2e test

### Does this close any open issues?
Please mention the issues here.

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/gitlab/gitlab.go [hunk 1]
```diff
@@ -30,11 +30,14 @@ var PluginEntry impl.Gitlab //nolint
 func main() {
 	gitlabCmd := &cobra.Command{Use: "gitlab"}
 	projectId := gitlabCmd.Flags().IntP("project-id", "p", 0, "gitlab project id")
+	connectionId := gitlabCmd.Flags().Uint64P("connection-id", "c", 0, "gitlab connection id")
```
Comment (likyh) [inline plugins/gitlab/gitlab.go:33]: we have some different styles of cmd flags. Such as: "connectionId", "connection", "connection-id".
Let's start a issue to talk about this.

File: config/config.go [hunk 2]
```diff
@@ -45,13 +45,12 @@ func initConfig(v *viper.Viper) {
 	v.SetConfigType("env")
 	envPath := getEnvPath()
 	// AddConfigPath adds a path for Viper to search for the config file in.
-	if envPath == "" {
```
Comment (likyh) [inline config/config.go:48]: I have no idea about this 😹

File: plugins/gitlab/impl/impl.go [hunk 3]
```diff
@@ -82,8 +85,23 @@ func (plugin Gitlab) PrepareTaskData(taskCtx core.TaskContext, options map[strin
...
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	err = connectionHelper.FirstById(connection, op.ConnectionId)
+	if err != nil {
+		return nil, err
+	}
 	apiClient, err := tasks.NewGitlabApiClient(taskCtx)
```
Comment (likyh) [inline plugins/gitlab/impl/impl.go:103]: Should use the connection to NewGitlabApiClient instead of using getConfig.

**Review by likyh — CHANGES_REQUESTED** [review]

File: plugins/gitlab/gitlab.go (hunk 1 above)
Comment (warren830) [reply plugins/gitlab/gitlab.go:33]: great！

File: plugins/gitlab/impl/impl.go (hunk 3 above)
Comment (likyh) [reply plugins/gitlab/impl/impl.go:103]: it's in NewGitlabApiClient. Or you can fix it in next pr.

Comment (likyh) [reply plugins/gitlab/impl/impl.go:103]: it's in NewGitlabApiClient. Or you can fix it in next pr.

**Review by likyh — APPROVED** [review]
LGTM but some issue will fix next PR

**Review by mindlesscloud — APPROVED** [review]
LGTM

**Pull request merged**